### PR TITLE
ref(txclusterer): Log Sentry message on missed projects for clustering

### PR DESCRIPTION
**Pull request by iker-barriocanal**
This PR logs a Sentry message when the clusterer misses some projects. This allows the creation of Sentry alerts for this specific use case. Projects should only remain unclustered when the task hits the soft limit timeout, and a `SoftTimeLimitExceeded` error should be emitted; however, this is not always the case and we sometimes have unclustered projects without such errors. Including the number of unclustered projects should also provide a bit more visibility of the impact of the issue.

### CODE REVIEW

File: src/sentry/ingest/transaction_clusterer/tasks.py [hunk 1]
```diff
@@ -102,6 +102,9 @@ def cluster_projects(projects: Sequence[Project]) -> None:
                 tags={"clustered": False},
                 sample_rate=1.0,
             )
+            sentry_sdk.capture_message(
+                f"Transaction clusterer missed projects: {unclustered}", level="error"
```
Comment (olksdr) [inline src/sentry/ingest/transaction_clusterer/tasks.py:106]: Can we keep the name  of the error always the same? 
```suggestion
                f"Transaction clusterer missed projects", level="error"
```

Which kind of makes it obvious what happens and use number of `unclustered` project as a additional data attached to the error?

File: src/sentry/ingest/transaction_clusterer/tasks.py [hunk 2]
```diff
@@ -102,6 +102,9 @@ def cluster_projects(projects: Sequence[Project]) -> None:
                 tags={"clustered": False},
                 sample_rate=1.0,
             )
+            sentry_sdk.set_tag("projs.total", len(projects))
+            sentry_sdk.set_tag("projs.unclustered", unclustered)
```
Comment (olksdr) [inline src/sentry/ingest/transaction_clusterer/tasks.py:106]: I'm not sure if we want to use tags for this,  I would rather use `set_extra` and would give it longer names, meaning `projs` -> `projects`

**Review by olksdr — APPROVED** [review]

File: src/sentry/ingest/transaction_clusterer/tasks.py (hunk 2 above)
Comment (iker-barriocanal) [reply src/sentry/ingest/transaction_clusterer/tasks.py:106]: Tags are queryable, so they allow us to evaluate how many projects were unclustered. Using the context would require opening all events individually and summing them up one by one. The number of unclustered projects is useful to evaluate the impact.

Comment (iker-barriocanal) [reply src/sentry/ingest/transaction_clusterer/tasks.py:106]: Regarding the tag names, I'm open to updating them but can make queries too long and is easy to understand as-is. What do you think?

Comment (olksdr) [reply src/sentry/ingest/transaction_clusterer/tasks.py:106]: I still prefer proper names, event if they are longer.

**Pull request merged**